### PR TITLE
Ensure history.replaceState is a function

### LINKED ISSUE #2488: TypeError: history.replaceState is not a function - in office.js
>> Please follow the issue template below. Failure to do so will result in a delay in answering your question.

## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.4.0`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

**Important**: Please fill in your exact version number above, e.g. `msal@1.1.3`.

## Framework

`webpack`, `react`

## Description

The redirect promise crashes due to `history.replaceState` is not a function.

## Error Message

history

## Security
- [ ] Is this issue security related?

## Regression
- [x] Did this behavior work before? 
Version: 2.3.1

## MSAL Configuration

No special configuration. Only using `clientId` and `cacheLocation: 'localStorage'`

## Reproduction steps

Just initialize an `msal` instance and use `msal.handleRedirectPromise()`.

## Expected behavior

`msal.handleRedirectPromise()` doesn't crash.

## Browsers/Environment
- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Safari
- [ ] IE
- [ ] Other (Please add browser name here)

### PR DESCRIPTION
**Pull request by jasonnutter**
Office.js will set `history.replaceState` to null, which satifies the existing `"replaceState" in history` check but causes a `history.replaceState` is not a function error.

### CODE REVIEW

File: change/@azure-msal-browser-2020-10-27-11-19-17-history-replace-state-function.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "patch",
+  "comment": "Ensure history.replaceState is a function",
+  "packageName": "@azure/msal-browser",
+  "email": "janutter@microsoft.com",
+  "dependentChangeType": "patch",
```
Comment (jasonnutter) [inline change/@azure-msal-browser-2020-10-27-11-19-17-history-replace-state-function.json:6]:
```suggestion
  "dependentChangeType": "none",
```

File: change/msal-2020-10-27-11-19-17-history-replace-state-function.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "patch",
+  "comment": "Ensure history.replaceState is a function",
+  "packageName": "msal",
+  "email": "janutter@microsoft.com",
+  "dependentChangeType": "patch",
```
Comment (jasonnutter) [inline change/msal-2020-10-27-11-19-17-history-replace-state-function.json:6]:
```suggestion
  "dependentChangeType": "none",
```

**Review by tnorling — APPROVED** [review]

**Review by pkanher617 — APPROVED** [review]

File: lib/msal-browser/src/utils/BrowserUtils.ts [hunk 3]
```diff
@@ -40,7 +40,8 @@ export class BrowserUtils {
      * Clears hash from window url.
      */
     static clearHash(): void {
-        if ("replaceState" in history) {
+        // Office.js sets history.replaceState to null
+        if (typeof history.replaceState === "function") {
```
Comment (ralphtheninja) [inline lib/msal-browser/src/utils/BrowserUtils.ts:44]: :+1: Maybe worth checking if there are more `in` checks in the code.

Comment (jasonnutter) [reply lib/msal-browser/src/utils/BrowserUtils.ts:44]: Don't see any others.

**Pull request merged**